cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java

```diff
@@ -149,6 +190,37 @@ public static DataprocClient fromConf(DataprocConf conf) throws IOException, Gen
     return new DataprocClient(new DataprocConf(conf, network, subnet), client, compute, useInternalIP);
   }
 
+  private static boolean isPeered(String systemNetwork, String systemProjectId, Network networkInfo,
+                                  boolean raiseException) {
+    List<NetworkPeering> peerings = networkInfo.getPeerings();
+    for (NetworkPeering peering : peerings) {
+      Matcher matcher = GCP_NETWORK_FULL_PATH.matcher(peering.getNetwork());
```

Comment:
It seems safer to get the system network URI and check if it is equal to the peering network rather than parsing this string. Can this be gotten from the Network object corresponding to the system network?